File: python/src/patterns/creational/factory/concrete_creators/csv_parser_creator.py
from src.patterns.creational.factory.abstract_creator.parser_creator import ParserCreator
from src.patterns.creational.factory.concrete_products.csv_parser import CsvParser
from src.patterns.creational.factory.product_interface.document_parser import DocumentParser

class CsvParserCreator(ParserCreator):
    def create_parser(self) -> DocumentParser:
        return CsvParser()

File: java/src/patterns/creational/factory/concretecreators/CsvParserCreator.java
package patterns.creational.factory.concretecreators;

import patterns.creational.factory.abstractcreator.ParserCreator;
import patterns.creational.factory.concreteproducts.CsvParser;
import patterns.creational.factory.productinterface.DocumentParser;

public class CsvParserCreator extends ParserCreator {
    @Override
    public DocumentParser createParser() {
        return new CsvParser();
    }
}
